apps/snitch_core/test/data/schema/product_property_test.exs

```diff
@@ -1,56 +1,105 @@
+      cs = ProductProperty.create_changeset(%ProductProperty{}, params)
+      assert cs.valid?
+    end
 
-  test "create unsuccessful if name empty", %{invalid_attrs: iva} do
-    %{valid?: validity} = ProductProperty.create_changeset(%ProductProperty{}, iva)
-    refute validity
-  end
+    test "fails with invalid params" do
+      cs = ProductProperty.create_changeset(%ProductProperty{}, %{})
+      refute cs.valid?
+      assert %{ product_id: ["can't be blank"],
+                property_id: ["can't be blank"],
+                value: ["can't be blank"],
+               } == errors_on(cs)
+    end
+
+    test "fails for duplicate product_id", context do
+      %{params: params} = context
+      cs = ProductProperty.create_changeset(%ProductProperty{}, params)
+      {:error, changeset} = Repo.insert(cs)
+      assert %{property_id: ["has already been taken"]} == errors_on(changeset)
+    end
+
+    test "fails for non-existent product_id, property_id" do
+      params = %{
+        product_id: -1,
+        property_id: 1,
+        value: "val"
+     }
+
+      cs = ProductProperty.create_changeset(%ProductProperty{}, params)
+      {:error, changeset} = Repo.insert(cs)
+      assert %{product_id: ["does not exist"]
+            } == errors_on(changeset)
+    end
 
-  test "update successfully", %{valid_attrs: va} do
-    cset = %{valid?: validity} = ProductProperty.create_changeset(%ProductProperty{}, va)
-    assert validity
-    assert {:ok, product_property} = Repo.insert(cset)
+    test "fails for non-existent property_id", context do
+      %{params: params} = context
+      params = %{
+        product_id: params.product_id,
+        property_id: -1,
+        value: "val"
+     }
 
-    params = %{value: "new value"}
-    cset = ProductProperty.update_changeset(product_property, params)
-    assert {:ok, new_product_property} = Repo.update(cset)
-    assert new_product_property.value != product_property.value
+      cs = ProductProperty.create_changeset(%ProductProperty{}, params)
+      {:error, changeset} = Repo.insert(cs)
+      assert %{property_id: ["does not exist"]
+            } == errors_on(changeset)
+    end
   end
 
-  test "update unsuccessful", %{valid_attrs: va, invalid_attrs: iva} do
-    cset = %{valid?: validity} = ProductProperty.create_changeset(%ProductProperty{}, va)
-    assert validity
-    assert {:ok, product_property} = Repo.insert(cset)
+  describe "update_changeset/2" do
+    setup do
+      product_property = insert(:product_property)
+
+      params = %{
+        product_id: product_property.product_id,
+        property_id: product_property.property_id,
+        value: product_property.value
+      }
+        [
+          changset:
+          %ProductProperty{}
+          |> ProductProperty.create_changeset(params)
+          |> apply_changes(),
+
+          params: params
+        ]
+    end
+
+    test "successfully with valid params", %{changset: product_property, params: params} do
```

Comment:
Here we should use already inserted Product Property for update changeset.
Also validate if the `update_changeset` has the changes we are about to update.

We are allowing to update the `property` and `value`. Both need to be tested.

Reply:
In update_changeset/2 , params given to create_changeset is already inserted in the database